### PR TITLE
{numlib}[GCC/12.3.0] beagle-lib v4.0.1 w/ CUDA 12.1.1

### PR DESCRIPTION
**Pull request by fizwit**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by migueldiascosta** [comment]
@boegelbot please test @ generoso

**Comment by migueldiascosta** [comment]
@boegelbot please test @ jsc-zen3

**Comment by migueldiascosta** [comment]
Test report by @migueldiascosta
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
gracehopper - Linux Rocky Linux 9.3, AArch64, ARM UNKNOWN (neoverse_v2), 1 x NVIDIA NVIDIA GH200 480GB, 550.54.15, Python 3.9.18
See https://gist.github.com/migueldiascosta/96342cba1e80044e8326ebaa84d6d05e for a full test report.

File: easybuild/easyconfigs/b/beagle-lib/beagle-lib-4.0.1-GCC-12.3.0-CUDA-12.1.1.eb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+easyblock = 'CMakeMake'
+
+name = 'beagle-lib'
+version = '4.0.1'
+versionsuffix = '-CUDA-%(cudaver)s'
+homepage = 'https://github.com/beagle-dev/beagle-lib'
```
Comment (migueldiascosta) [inline easybuild/easyconfigs/b/beagle-lib/beagle-lib-4.0.1-GCC-12.3.0-CUDA-12.1.1.eb:6]: @fizwit sorry for nitpicking, but a blank line before `homepage` does make it easier to read, do you mind adding that?

Comment (verdurin) [reply easybuild/easyconfigs/b/beagle-lib/beagle-lib-4.0.1-GCC-12.3.0-CUDA-12.1.1.eb:7]: @migueldiascosta I went ahead and made that change myself.

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
easybuild-el8.cloud.in.bmrc.ox.ac.uk - Linux Rocky Linux 8.10, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/verdurin/feba5d52ff991a40af6de11138c3b7c2 for a full test report.

**Review by verdurin — APPROVED** [review]

**Pull request merged**